hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/BaseHoodieWriteClient.java

```diff
@@ -236,8 +236,8 @@ public boolean commitStats(String instantTime, HoodieData<WriteStatus> writeStat
       commit(table, commitActionType, instantTime, metadata, stats, writeStatuses);
       postCommit(table, metadata, instantTime, extraMetadata);
       LOG.info("Committed " + instantTime);
-      releaseResources(instantTime);
     } catch (IOException e) {
+      releaseResources(instantTime);
```

Comment:
why not move releaseResources to finally block ? and so we can avoid L 260 right ?

Reply:
we want to release resources after table services

Reply:
gotcha. then shouldn't we try to do the same at L255

Reply:
makes sense